packages/@react-aria/overlays/src/useOverlay.ts

```diff
@@ -111,6 +111,7 @@ export function useOverlay(props: OverlayProps, ref: RefObject<HTMLElement>): Ov
   // Handle the escape key
   let onKeyDown = (e) => {
     if (e.key === 'Escape' && !isKeyboardDismissDisabled) {
+      e.stopPropagation();
```

Comment:
this seems separate from the rest, less controversial, and higher priority. should this just be its own PR to get in easily?

Reply:
See https://github.com/adobe/react-spectrum/pull/2976